hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/table/HoodieFlinkTable.java

```diff
@@ -110,16 +109,15 @@ protected HoodieIndex getIndex(HoodieWriteConfig config, HoodieEngineContext con
   @Override
   public <T extends SpecificRecordBase> Option<HoodieTableMetadataWriter> getMetadataWriter(Option<T> actionMetadata) {
     synchronized (this) {
-      if (!isMetadataAvailabilityUpdated) {
-        // This code assumes that if metadata availability is updated once it will not change.
-        // Please revisit this logic if that's not the case. This is done to avoid repeated calls to fs.exists().
+      // Metadata table bootstrapping might have failed due to pending actions.
```

Comment:
this method in geneneral seems like a duplicated code, Can we move this to the super class